src/components/src/main/java/org/apache/jmeter/extractor/RegexExtractor.java

```diff
@@ -178,6 +182,14 @@ private void extractWithOroRegex(SampleResult previousResult, JMeterVariables va
         }
     }
 
+    private void failResult(SampleResult previousResult){
+        previousResult.setSuccessful(false);
+        AssertionResult res = new AssertionResult(getName());
+        res.setFailure(true);
+        res.setFailureMessage("Pattern not found: " + getRegex());
```

Comment:
I suggest including the "source" that was checked against the regexp. In other words, something like `Pattern ... not found in ...<response body | ... >"